.github/workflows/php.yml

```diff
@@ -0,0 +1,49 @@
+name: Poser
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container: pugx/poser:php80
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Validate composer.json and composer.lock
+        run: composer validate
+
+      - name: Cache Composer packages
+        id: composer-cache
+        uses: actions/cache@v2
+        with:
+          path: vendor
+          key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-php-
+
+      - name: Install dependencies
+        uses: php-actions/composer@v5
+        with:
+          args: -n --no-progress --no-suggest
+          composer_version: 2
+
+      - name: Coding Standard Checks
+        uses: docker://oskarstark/php-cs-fixer-ga
+        with:
+          args: fix --verbose --diff --dry-run
+
+      - name: phpspec
+        uses: php-actions/phpspec@master
+        with:
+          config: phpspec.yml
+
+      - name: Run tests (phpspec)
+        run: ./bin/phpspec run --format=pretty
+
+      - name: Run coverage tests (phpspec)
+        run: ./bin/phpspec run -f progress -c phpspec-coverage.yml
+
+      - name: Run behat tests
+        run: ./bin/behat
```

Comment:
maybe the jobs for php74 are missing?

Reply:
Yes, I am thinking on what is the best approach but I think that the correct way is to use a PHP version matrix